commit 2e8cdc12c7b153b5cc971d4b8bfd1d0b69cb40b6
Author: YYS <nickey.yang@rock-chips.com>
Date:   Tue Mar 21 16:27:03 2017 +0800

    drm/mediatek: fix mtk_hdmi_setup_vendor_specific_infoframe mistake
    
    commit 014580ffab654bb83256783a2b185cf6c06dffaa upstream.
    
    mtk_hdmi_setup_vendor_specific_infoframe will return before handle
    mtk_hdmi_hw_send_info_frame.Because hdmi_vendor_infoframe_pack
    returns the number of bytes packed into the binary buffer or
    a negative error code on failure.
    So correct it.
    
    Fixes: 8f83f26891e1 ("drm/mediatek: Add HDMI support")
    Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
    Signed-off-by: CK Hu <ck.hu@mediatek.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/gpu/drm/mediatek/mtk_hdmi.c b/drivers/gpu/drm/mediatek/mtk_hdmi.c
index 0e8c4d9..e097780 100644
--- a/drivers/gpu/drm/mediatek/mtk_hdmi.c
+++ b/drivers/gpu/drm/mediatek/mtk_hdmi.c
@@ -1061,7 +1061,7 @@ static int mtk_hdmi_setup_vendor_specific_infoframe(struct mtk_hdmi *hdmi,
 	}
 
 	err = hdmi_vendor_infoframe_pack(&frame, buffer, sizeof(buffer));
-	if (err) {
+	if (err < 0) {
 		dev_err(hdmi->dev, "Failed to pack vendor infoframe: %zd\n",
 			err);
 		return err;
